tests/integration/test_data/gomod_e2e_test/container/Containerfile-fail

```diff
@@ -0,0 +1,15 @@
+FROM docker.io/golang:1.18.1 AS build
+
+# Network access should be disabled for testing Cachi2 functionality
+RUN curl www.google.com
```

Comment:
When testing locally, I was getting this error when the build had network access:
> `FAILED tests/integration/test_package_managers.py::test_e2e[gomod_e2e_test] - UnicodeDecodeError: 'utf-8' codec can't decode byte 0xa0 in position 10810: invalid start byte`

Maybe we could just fetch the header and suppress the other output? Something like `curl -IsS www.google.com`?

Reply:
+1

Also, do we need a second Containerfile for this? How about checking it in the successful build:
```Dockerfile
RUN if curl -IsS www.google.com; then echo "Has network access!"; exit 1; fi
```

Reply:
Great idea. changed